gdal/frmts/hfa/hfadataset.cpp

```diff
@@ -3007,8 +3043,13 @@ CPLErr HFARasterBand::SetDefaultRAT( const GDALRasterAttributeTable * poRAT )
     if( poRAT == nullptr )
         return CE_Failure;
 
-    poDefaultRAT = poRAT->Clone();
-    return WriteNamedRAT("Descriptor_Table", poRAT);
+    poDefaultRAT = nullptr;
```

Comment:
delete poDefaultRAT missing before nullifying it

Reply:
bad patch cleanup, will fix.